### DISCUSSION TITLE
What the difference between auto-removal of jobs in queues and in wokers

### DISCUSSION THREAD

**Discussion by larryro** [taskforcesh/bullmq#2383] [Q&A]
From the description, we can see that both are used to set the processing strategy after a job fails or succeeds. If these two settings are inconsistent, which one has a higher priority?

https://docs.bullmq.io/guide/queues/auto-removal-of-jobs
https://docs.bullmq.io/guide/workers/auto-removal-of-jobs

![image](https://github.com/taskforcesh/bullmq/assets/5787117/191ac541-97ae-43e7-8b9b-875fe12060f8)
![image](https://github.com/taskforcesh/bullmq/assets/5787117/2b9acc09-5aeb-4534-b48f-12e6af37f3e2)

**Comment by rosslavery** [comment]
My understanding is that the "worker level" options can be thought of as "default job options" and the job level ones are "job/queue options". 

So the job-level ones take precedence over the worker-level ones.
